msnc/model.py

```diff
@@ -75,35 +71,66 @@ def _init_linears(self, linear_params):
             linears.append(linear)
         return nn.ModuleList(linears)
 
-    def run_training(self, output_dir, training_set, development_set=None):
+    def run_training(
+        self,
+        output_dir,
+        training_set,
+        development_set=None,
+        test_set=None,
```

Comment:
It's not a good idea to give a test set to a training method because of the following three reasons:

- A. the test set is essentially unnecessary for the training.
- B. it's easy to see the accuracies on the test set and tune our model according to the results.
- C. we may mistake the test set for the training or development sets when we change the codes. 🐛 

I can understand what you want to do. So, my idea is as follows:

- a. use the original code and select the best model on the development set for the test. This solves the above three issues.
- b. define a child class (e.g., `ModelForExperiment`) that has this `run_training` method with the `test_set` argument and use it for experiments. This slightly solves the issues A and C.

Reply:
It's OK for me and I understand your opinion.
I revised my PR so that `run_training` does not take `test_set` and
take `save_best_model` to save `best.model` for testing.
What do you think about this change? :eyes:

Reply:
looks great for me 👍